src/backend/distributed/commands/multi_copy.c

```diff
@@ -987,25 +863,13 @@ OpenCopyTransactions(CopyStmt *copyStatement, ShardConnections *shardConnections
 		ShardPlacement *placement = (ShardPlacement *) lfirst(placementCell);
 		char *nodeName = placement->nodeName;
 		int nodePort = placement->nodePort;
-		WorkerNode *workerNode = FindWorkerNode(nodeName, nodePort);
-		int workerGroupId = 0;
 		char *nodeUser = CurrentUserName();
-		PGconn *connection = ConnectToNode(nodeName, nodePort, nodeUser);
-
-		TransactionConnection *transactionConnection = NULL;
+		MultiConnection *connection = NULL;
+		uint32 connectionFlags = SESSION_LIFESPAN | FOR_DML;
```

Comment:
It basically means a single COPY would leave dozens of connections to workers open until the end of the session, which will likely cause failures and mess up pgbouncer setups like Citus Cloud. This should only ever be optional.